ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/sendtransaction/RetryMechanism.java

```diff
@@ -15,11 +15,23 @@
 import io.vertx.core.buffer.Buffer;
 import io.vertx.core.http.HttpClientResponse;
 
-public interface RetryMechanism<T> {
+public abstract class RetryMechanism<T> {
 
-  class RetryException extends RuntimeException {};
+  private final int maxRetries;
+  private int retriesPerformed = 0;
 
-  boolean mustRetry(final HttpClientResponse response, final Buffer body);
+  public RetryMechanism(int maxRetries) {
+    this.maxRetries = maxRetries;
+  }
 
-  void retry(final T context, Runnable sender) throws RetryException;
```

Comment:
As this was the only line in `RetryMechanism` using generics, you can now remove it from the class 🙂

Reply:
Done.